presto-main/src/main/java/com/facebook/presto/operator/SettableLookupSourceSupplier.java

```diff
@@ -46,10 +46,12 @@ public SettableLookupSourceSupplier(List<Type> types)
         return types;
     }
 
+    @SuppressWarnings("RedundantCast")
     @Override
     public ListenableFuture<LookupSource> getLookupSource(OperatorContext operatorContext)
     {
-        return transform(lookupSourceFuture, (AsyncFunction<SharedLookupSource, LookupSource>) Futures::immediateFuture);
+        // The Eclipse compiler requires the explicit <LookupSource> cast.
+        return transform(lookupSourceFuture, (AsyncFunction<SharedLookupSource, LookupSource>) Futures::<LookupSource>immediateFuture);
```

Comment:
IDEA thinks the cast is redundant after adding the type argument (hence the suppression you added). Without it, the Eclipse compiler seems to crash:
```
Error:java: Internal compiler error: java.lang.NullPointerException at org.eclipse.jdt.internal.compiler.lookup.InferenceContext18.checkExpression(InferenceContext18.java:796)
```
This sees to be https://bugs.eclipse.org/bugs/show_bug.cgi?id=438945